### PR TITLE
Display battle information and implement current health tracking

### LINKED ISSUE #14: Display and update ship health
When you add a ship its health should be equal to the strength.

After battles the health of the ship should be updated to the DB

if it needs maintenance is based on health from DB
- Add a maxHealth column
- Add a currentHealth column to determine if a ship is combat ready after a battle rather than the current Rand function.
     - This will show data persistence

### PR DESCRIPTION
**Pull request by ljweston**
Closes #14

### CODE REVIEW

File: Model/abstractShip.php [hunk 1]
```diff
@@ -93,14 +99,14 @@ public function setWeaponPower($weaponPow)
         $this->weaponPower = $weaponPow;
     }
 
-    public function setStrength($strength)
+    public function setMaxHealth($maxHealth)
     {
-        $this-> strength = $strength;
+        $this-> maxHealth = $maxHealth;
```
Comment (mlellis) [inline Model/abstractShip.php:104]: Remove the space between $this-> and maxHealth

File: manage/ships/repair.php [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+use Service\Container;
+
+$id = isset($_GET['id']) ? $_GET['id'] : null;
+
+$container = new Container($configuration);
+$shipLoader = $container->getShipLoader();
+// get our ship
+$ship = $shipLoader->findOneById($id);
+$ship->repairShip(); // updated currenthealth on the ship obj
```
Comment (mlellis) [inline manage/ships/repair.php:13]: This can now be removed. See note in PdoShipStorage

File: manage/ships/repair.php [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+$shipLoader->repairShip($ship);
+
+header('Location: /manage/ships/index.php');
```
Comment (mlellis) [inline manage/ships/repair.php:16]: I think it would make sense to redirect to the ship view page

**Review by mlellis — CHANGES_REQUESTED** [review]

File: Model/abstractShip.php [hunk 4]
```diff
@@ -42,23 +48,23 @@ public function getNameAndSpecs($useShortFormat = false)
...
                 $this->weaponPower,
                 $this->getJediFactor(),
-                $this->strength
+                $this->maxHealth
             );
         }
     }
 
-    public function givenShipMoreStrength($givenShip)
+    public function givenShipMoreHealth($givenShip)
```
[the remark this answers is not in the document]
Comment (ljweston) [reply Model/abstractShip.php:64]: Must have been from an earlier tutorial. Removed.

File: Service/pdoShipStorage.php [hunk 5]
```diff
@@ -57,34 +57,48 @@ public function saveShip(AbstractShip $ship)
...
 
         $statement->execute();
 
         // maybe return errors if any ecountered
     }
 
+    public function repairShip(AbstractShip $ship)
+    {
+        $pdo = $this->pdo;
+        $query = 'UPDATE OOPShips.ship SET current_health = :currentHealthVal WHERE id = :idVal';
```
[the remark this answers is not in the document]
Comment (ljweston) [reply Service/pdoShipStorage.php:79]: Done, but kept the $ship object being passed in so I could access the ID of the ship

File: Model/abstractShip.php [hunk 6]
```diff
@@ -29,6 +30,11 @@ public static function getTeams()
         return [self::EMPIRE, self::REBEL];
     }
 
+    public function repairShip()
```
[the remark this answers is not in the document]
Comment (ljweston) [reply Model/abstractShip.php:33]: Removed

File: index.php [hunk 7]
```diff
@@ -101,7 +101,7 @@
                             <?php endforeach; ?>
                         </select>
                         <br>
-
+                        <?php var_dump(BattleManager::TYPE_NORMAL)?>
```
[the remark this answers is not in the document]
Comment (ljweston) [reply index.php:104]: Removed

File: Model/abstractShip.php (hunk 1 above)
Comment (ljweston) [reply Model/abstractShip.php:104]: Done

File: manage/ships/repair.php (hunk 2 above)
Comment (ljweston) [reply manage/ships/repair.php:13]: repairShip Removed

File: manage/ships/repair.php (hunk 3 above)
Comment (ljweston) [reply manage/ships/repair.php:16]: Done

File: Model/rebelShip.php [hunk 8]
```diff
@@ -20,7 +20,14 @@ public function getType()
 
     public function isFunctional()
     {
-        return true;
+        /**
+         * check current health of the ship parent::getCurrentHealth()
+         */
+        if (parent::getCurrentHealth() <= ((10/100) * parent::getMaxHealth())) {
+            return false;
+        } else {
```
[the remark this answers is not in the document]
Comment (mlellis) [reply Model/rebelShip.php:28]: You used $this, but you didn't update the logic.

File: Model/ship.php [hunk 9]
```diff
@@ -22,7 +22,7 @@ public function getType()
     public function isFunctional()
     {
         // check that the currentHealth is >= 15% of the maxHealth
-        if (parent::getCurrentHealth() <= ((20/100) * parent::getMaxHealth())) {
+        if ($this->getCurrentHealth() <= ((20/100) * $this->getMaxHealth())) {
```
Comment (mlellis) [inline Model/ship.php:25]: You used $this, but you didn't update the logic.

File: Service/pdoShipStorage.php [hunk 10]
```diff
@@ -73,14 +73,13 @@ public function saveShip(AbstractShip $ship)
         // maybe return errors if any ecountered
     }
 
-    public function repairShip(AbstractShip $ship)
+    public function repairShip($id)
```
Comment (mlellis) [inline Service/pdoShipStorage.php:76]: Why did you remove passing in the object here? I think that is still needed.

File: Service/shipStorageInterface.php [hunk 11]
```diff
@@ -34,7 +34,7 @@ public function saveShip(AbstractShip $newShip);
 
     public function updateShip(AbstractShip $ship);
 
-    public function repairShip(AbstractShip $ship);
+    public function repairShip($id);
```
Comment (mlellis) [inline Service/shipStorageInterface.php:37]: Should still pass object.

File: Service/shipLoader.php [hunk 12]
```diff
@@ -57,11 +57,11 @@ public function updateShip(AbstractShip $ship)
         $this->shipStorage->updateShip($ship);
     }
 
-    public function repairShip(AbstractShip $ship)
+    public function repairShip($id)
```
Comment (mlellis) [inline Service/shipLoader.php:60]: Should still pass object.

File: manage/ships/repair.php [hunk 13]
```diff
@@ -8,9 +8,7 @@
 
 $container = new Container($configuration);
 $shipLoader = $container->getShipLoader();
-// get our ship
-$ship = $shipLoader->findOneById($id);
-$ship->repairShip(); // updated currenthealth on the ship obj
-$shipLoader->repairShip($ship);
 
-header('Location: /manage/ships/index.php');
+$shipLoader->repairShip($id);
```
Comment (mlellis) [inline manage/ships/repair.php:12]: Should still pass in the object.

Also might be good to show an error message if the ship is not found.

**Review by mlellis — CHANGES_REQUESTED** [review]

File: Model/ship.php (hunk 9 above)
Comment (ljweston) [reply Model/ship.php:25]: Fixed the logic to be simplified

File: Service/pdoShipStorage.php (hunk 10 above)
Comment (ljweston) [reply Service/pdoShipStorage.php:76]: Reverted changes.

File: Service/shipStorageInterface.php (hunk 11 above)
Comment (ljweston) [reply Service/shipStorageInterface.php:37]: Reverted changes

File: Service/shipLoader.php (hunk 12 above)
Comment (ljweston) [reply Service/shipLoader.php:60]: Passes object again

File: manage/ships/repair.php (hunk 13 above)
Comment (ljweston) [reply manage/ships/repair.php:12]: Added logic to return to the view page without an ID and display "ship not found" and allows user to go back to previous pages.

File: manage/ships/repair.php [hunk 14]
```diff
@@ -8,7 +8,10 @@
 
 $container = new Container($configuration);
 $shipLoader = $container->getShipLoader();
-
-$shipLoader->repairShip($id);
+$ship = $shipLoader->findOneById($id);
+if ($ship === null) {
+    header('Location: /manage/ships/view.php');
```
Comment (mlellis) [inline manage/ships/repair.php:13]: Did you test this logic? This is trying to redirect to the view page with no id. Also, I think it will fall through this conditional because there is no `die;`. So then it will call the `repairShip` method below.

I think it would be better to just display an error message here anyway.

**Review by mlellis — CHANGES_REQUESTED** [review]

**Review by mlellis — APPROVED** [review]
